rllib/core/rl_trainer/tests/test_trainer_runner.py

```diff
@@ -148,6 +141,60 @@ def test_add_remove_module(self):
                 set(result["loss"]) - {"total_loss"}, module_ids_before_add
             )
 
+    def test_trainer_runner_no_gpus(self):
+        env = gym.make("CartPole-v1")
+        trainer_class = BCTfRLTrainer
+        trainer_cfg = dict(
+            module_class=DiscreteBCTFModule,
+            module_kwargs={
+                "observation_space": env.observation_space,
+                "action_space": env.action_space,
+                "model_config": {"hidden_dim": 32},
+            },
+            optimizer_config={"lr": 1e-3},
+            in_test=True,
+        )
+        runner = TrainerRunner(trainer_class, trainer_cfg, num_gpus=0)
+
+        local_trainer = trainer_class(**trainer_cfg)
+        local_trainer.build()
+
+        # make the state of the trainer and the local runner identical
+        local_trainer.set_state(runner.get_state()[0])
+
+        reader = get_cartpole_dataset_reader(batch_size=500)
+        batch = reader.next()
+        batch = batch.as_multi_agent()
+        check(local_trainer.update(batch), runner.update(batch)[0])
+
+        new_module_id = "test_module"
+
+        # add a test_module
+        runner.add_module(
+            module_id=new_module_id,
+            module_cls=DiscreteBCTFModule,
+            module_kwargs={
+                "observation_space": env.observation_space,
+                "action_space": env.action_space,
+                "model_config": {"hidden_dim": 32},
+            },
+            optimizer_cls=tf.keras.optimizers.Adam,
+        )
+        local_trainer.add_module(
+            module_id=new_module_id,
+            module_cls=DiscreteBCTFModule,
+            module_kwargs={
+                "observation_space": env.observation_space,
+                "action_space": env.action_space,
+                "model_config": {"hidden_dim": 32},
+            },
+            optimizer_cls=tf.keras.optimizers.Adam,
+        )
+
+        # make the state of the trainer and the local runner identical
+        local_trainer.set_state(runner.get_state()[0])
+        check(local_trainer.get_state(), runner.get_state()[0])
```

Comment:
do another update with the new module and the states? there is no point in checking the states if you made them the same just in a line before.